Recruit-Training/Advanced-Recruit-Training/Embedded-Teaser/Training/Core/Src/Training.c

```diff
@@ -1,6 +1,60 @@
 #include "Training.h"
+/*
+    CONDITIONS FOR FAILURE
+    
+    -the velocity and direction bit don't express the same direction
+    -the input for the first and second motor aren't EXACTLY the same
+    -more than one headlight choice is chosen
+    -both left and right signals are on at the same time
 
+*/
 void trainingTask(uint8_t* data)
 {
+    uint8_t tempVelocityMotor1 = data[0] >> 6;
+    tempVelocityMotor1 = tempVelocityMotor1 << 7;
 
+    uint8_t tempVelocityMotor2 = data[0] >> 6;
+    tempVelocityMotor2 = tempVelocityMotor2 << 7;
```

Comment:
Are you trying to get the velocity itself or the sign? Currently all you've done is made the sign of the velocity the MSB of the variables and the rest of the bits are 0.

Reply:
Oh I think I understand what you're doing. Looked a bit further down. Can you write a comment to explain it a bit on this line?